### PR TITLE
added implementation

### PR DESCRIPTION
**Pull request by boroda4436**

### CODE REVIEW

File: result_file.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Path,Quantity
```
Comment (roman-mate) [inline result_file.txt]: maybe use filanames: input.txt, output.txt?

File: src/main/java/core/basesyntax/Main.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
+package core.basesyntax;
```
Comment (roman-mate) [inline src/main/java/core/basesyntax/Main.java]: why packagename basesyntax?

File: src/main/java/core/basesyntax/Main.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    public static void main(String[] args) {
+
+        if (args.length < 2) {
+            System.err.println("Invalid input params. Expecting to receive: "
+                    + "source_directories.txt result_file.txt (where the result will be written)");
+            return;
+        }
+
+        String input = args[0];
+        String output = args[1];
```
Comment (roman-mate) [inline src/main/java/core/basesyntax/Main.java]: should we recommend smth like this? 
https://stackoverflow.com/questions/55868420/how-to-get-command-line-arguments-in-spring-boot

File: src/main/java/core/basesyntax/model/DirectoryData.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.model;
+
+import java.util.List;
+
+public class DirectoryData {
+
+    private String directory;
+    private List<String> data;
```
Comment (roman-mate) [inline src/main/java/core/basesyntax/model/DirectoryData.java]: suggestion: subdirectories?

File: src/main/java/core/basesyntax/Main.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import core.basesyntax.service.WriterService;
+import core.basesyntax.util.CsvUtil;
+import core.basesyntax.util.DisplayResult;
+
+import java.io.File;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Main {
+    public static void main(String[] args) {
```
Comment (roman-mate) [inline src/main/java/core/basesyntax/Main.java]: this method is way too long.

File: src/main/java/core/basesyntax/model/DirectoryWalker.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    @Override
+    public DirectoryData call() {
+        try {
+            result = Files.walk(Paths.get(dir))
+                    .parallel()
+                    .filter(p -> !p.toFile().isDirectory())
+                    .map(Path::toString)
+                    .collect(Collectors.toList());
+        } catch (IOException e) {
+            System.err.println(e.getMessage());
```
Comment (roman-mate) [inline src/main/java/core/basesyntax/model/DirectoryWalker.java]: bad practice. always rethrow an exception.

File: src/main/java/core/basesyntax/service/FileReaderService.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileReaderService {
+
+    public static List<String> parseFile(String file) {
```
Comment (roman-mate) [inline src/main/java/core/basesyntax/service/FileReaderService.java]: service with static method looks weird.

File: src/main/java/core/basesyntax/util/CsvUtil.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.util;
+
+import core.basesyntax.model.DirectoryData;
+
+public class CsvUtil {
```
Comment (roman-mate) [inline src/main/java/core/basesyntax/util/CsvUtil.java]: why not some library to write csv file?

File: src/main/java/core/basesyntax/model/DirectoryData.java (hunk 4 above)
Comment (boroda4436) [reply src/main/java/core/basesyntax/model/DirectoryData.java]: what about files? we can have both of them

File: src/main/java/core/basesyntax/service/FileReaderService.java (hunk 7 above)
Comment (boroda4436) [reply src/main/java/core/basesyntax/service/FileReaderService.java]: forgot to remove this. Thanks!

File: src/main/java/core/basesyntax/util/CsvUtil.java (hunk 8 above)
Comment (boroda4436) [reply src/main/java/core/basesyntax/util/CsvUtil.java]: I think this is not required for this case

Comment (boroda4436) [reply src/main/java/core/basesyntax/util/CsvUtil.java]: we will have a separate reference project focused on CSV works

File: src/main/java/core/mate/model/DirectoryWalker.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    @Override
+    public DirectoryData call() {
+        try {
+            result = Files.walk(Paths.get(dir))
+                    .parallel()
+                    .filter(p -> !p.toFile().isDirectory())
+                    .map(Path::toString)
+                    .collect(Collectors.toList());
+        } catch (IOException e) {
+            System.err.println(e.getMessage());
```
Comment (roman-mate) [inline src/main/java/core/mate/model/DirectoryWalker.java]: why is this needed? when app breaks, stack trace will be printed to the console. if we println, some info will be printed twice?

File: src/main/java/core/mate/service/DirectoryWalkerService.java [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+        for (String directory : directories) {
+            DirectoryWalker directoryWalkerThread = new DirectoryWalker(directory);
+            threads.add(directoryWalkerThread);
+        }
+
+        List<Future<DirectoryData>> futures;
+        try {
+            futures = executorService.invokeAll(threads);
+        } catch (InterruptedException e) {
```
Comment (roman-mate) [inline src/main/java/core/mate/service/DirectoryWalkerService.java:29]: why ignore exception?

File: src/main/java/core/mate/service/DirectoryWalkerService.java [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+            return Collections.emptyList();
+        }
+
+        List<DirectoryData> results = new ArrayList<>();
+        for (Future<DirectoryData> future : futures) {
+            try {
+                results.add(future.get());
+            } catch (InterruptedException | ExecutionException e) {
+                e.printStackTrace();
```
Comment (roman-mate) [inline src/main/java/core/mate/service/DirectoryWalkerService.java:39]: why ignore exception?

File: src/main/java/core/mate/util/CmdArgsParser.java [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+        Option output = new Option("o", "output", true, "output file");
+        output.setRequired(true);
+        options.addOption(output);
+
+        CommandLine cmd;
+        try {
+            CommandLineParser parser = new DefaultParser();
+            cmd = parser.parse(options, args);
+        } catch (ParseException e) {
+            System.out.println(e.getMessage());
```
Comment (roman-mate) [inline src/main/java/core/mate/util/CmdArgsParser.java]: we should never use System.out.println in exception handling/logging.

File: src/main/java/core/basesyntax/model/DirectoryData.java (hunk 4 above)
Comment (roman-mate) [reply src/main/java/core/basesyntax/model/DirectoryData.java]: agree. let's keep it as data.

File: src/main/java/core/mate/util/CmdArgsParser.java [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+import org.apache.commons.cli.ParseException;
+
+public class CmdArgsParser {
+    private final CommandLine cmd;
+
+    public CmdArgsParser(String[] args) {
+        this.cmd = parse(args);
+    }
+
+    CommandLine parse(String[] args) throws ArgumentsParseException {
```
Comment (Sofasmile) [inline src/main/java/core/mate/util/CmdArgsParser.java:20]: access modifier?

File: src/test/java/core/mate/util/CmdArgsParserTest.java [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+
+    @Test
+    public void parseArgsOk() {
+        String[] args = {"--input=input.txt", "--output=output.txt"};
+        cmdArgsParser = new CmdArgsParser(args);
+        cmdArgsParser.parse(args);
+    }
+
+
+}
```
Comment (Sofasmile) [inline src/test/java/core/mate/util/CmdArgsParserTest.java]: add empty line

File: src/main/java/core/mate/util/CmdArgsParser.java (hunk 13 above)
Comment (boroda4436) [reply src/main/java/core/mate/util/CmdArgsParser.java:20]: default access modifier
